plasma_framework/contracts/src/utils/FailFastReentrancyGuard.sol

```diff
@@ -0,0 +1,21 @@
+pragma solidity 0.5.11;
+
+/**
+ * @notice Reentrancy guard that fails immediately when a reentrace occurs
+ * @dev Instead of using the one from openzepplin for the following reasons:
```

Comment:
@dev We use a simple mutex instead of https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol for the following reasons:

Reply:
done